devicetypes/stelpro/stelpro-maestro-thermostat.src/stelpro-maestro-thermostat.groovy

```diff
@@ -349,21 +356,50 @@ def updateWeather() {
 	// If there is a zipcode defined, weather forecast will be sent. Otherwise, no weather forecast.
 	if (settings.zipcode) {
 		log.debug "ZipCode: ${settings.zipcode}"
-		weather = getTwcConditions(settings.zipcode)
+		try {
+			// If we do not have a zip-code setting we've determined as invalid, try to use the zip-code defined.
+			if (!state.invalidZip) {
+				weather = getTwcConditions(settings.zipcode)
+			}
+		} catch (e) {
+			log.debug "getTwcConditions exception: $e"
+			// There was a problem obtaining the weather with this zip-code, so fall back to the hub's location and note this for future runs.
+			state.invalidZip = true
+		}
 
-		// Check if the variable is populated, otherwise return.
 		if (!weather) {
-			log.debug("Something went wrong, no data found.")
-			return false
+			try {
+				// It is possible that a non-U.S. zip-code was used, so try with the location's lat/lon.
+				if (location?.latitude && location?.longitude) {
+					// Restrict to two decimal places for the API
+					java.text.DecimalFormat formatter = new java.text.DecimalFormat()
+					formatter.setMaximumFractionDigits(2)
+					formatter.setMinimumFractionDigits(0)
+
+					def lat = formatter.format(location.latitude)
+					def lon = formatter.format(location.longitude)
+
+					def latlon = "${lat},${lon}"
+					weather = getTwcConditions(latlon)
```

Comment:
`weather = getTwcConditions(sprintf("%.2f, %.2f", lat, lon)`

Reply:
I'll give this a test. Thanks